storm-core/src/jvm/backtype/storm/scheduler/TopologyDetails.java

```diff
@@ -403,13 +415,23 @@ public void addResourcesForExec(ExecutorDetails exec, Map<String, Double> resour
      * Add default resource requirements for a executor
      */
     public void addDefaultResforExec(ExecutorDetails exec) {
+        Double topologyComponentCpuPcorePercent = Utils.getDouble(topologyConf.get(Config.TOPOLOGY_COMPONENT_CPU_PCORE_PERCENT), null);
+        if (topologyComponentCpuPcorePercent == null) {
+            LOG.warn("default value for topology.component.cpu.pcore.percent needs to be set!");
```

Comment:
Why don't we throw something that's not an NPE here, then? I'd rather fail earlier than later.